crates/tiny/src/main.rs

```diff
@@ -178,3 +193,41 @@ fn run(
 
     runtime.block_on(local);
 }
+
+// Helper to parse SASL config into a libtiny_client SASL struct
+fn sasl_from_config(tls: bool, sasl_config: &config::SASLAuth) -> Result<SASLAuth, String> {
+    match sasl_config {
+        config::SASLAuth::Plain { username, password } => Ok(SASLAuth::Plain {
+            username: username.to_string(),
+            password: password.to_string(),
```

Comment:
```suggestion
            username: username.clone(),
            password: password.clone(),
```

`to_string` reads like you're converting some type other than string/str to `String`, but this is cloning a borrowed `String`. `clone` makes this more clear IMO.